### PR TITLE
[feat] Add text line styles

### PR DESCRIPTION
**Pull request by htin1**
Screenshot 2026-07-16 at 2 29 28 PM

## Summary

- Add underline, strikethrough, and overline controls to text styling.
- Persist the three traits and render them across static, per-word, and typewriter text.
- Expose matching `underline`, `strikethrough`, and `overline` booleans to agent text tools.

## Why

Text layers should support the familiar line-style controls available in other video editors, with the same capabilities available to the agent.

## Impact

Editors can apply lines below, through, or above text from the inspector. Existing projects remain compatible because absent fields decode as disabled.

## Validation

- `swift test` — 1,063 tests passed

---

> [!NOTE]
> **Low Risk**
> Localized text styling and rendering changes with backward-compatible defaults; no auth, data migration, or critical-path risk beyond text compositing edge cases.
> 
> **Overview**
> Adds **underline**, **strikethrough**, and **overline** as first-class text style traits end to end.
> 
> `TextStyle` gains three booleans (default off, tolerant decode for older projects). Underline and strikethrough map into Core Text attributed-string attributes; **overline** is drawn manually in `TextFrameRenderer` on static, typewriter, and per-word animation paths so it still shows when text is revealed word-by-word.
> 
> The inspector **Style** row gets toggle buttons for all three (line traits commit without refitting the text box). Agent `add_texts` / `update_text` style patches accept `underline`, `strikethrough`, and `overline` booleans, wired through parsing and `applyTextStylePatch`. Tests cover agent application, project round-trip, and a rendering smoke check.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 78df624970c5e923064b13a03e44fc544889f778. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: Sources/PalmierPro/Compositing/TextFrameRenderer.swift [hunk 1]
```diff
@@ -187,6 +214,7 @@ enum TextFrameRenderer {
                 }
                 ctx.textPosition = CGPoint(x: penX, y: penY)
                 CTLineDraw(wordLine, ctx)
+                if style.isOverlined, let font { drawOverline(ctx, x: penX, y: penY, width: wWidth, font: font, color: st.color) }
```
[the remark this answers is not in the document]
Comment (htin1) [reply Sources/PalmierPro/Compositing/TextFrameRenderer.swift:217]: that's expected.
Screenshot 2026-07-16 at 2 43 04 PM
